### ISSUE TITLE
`useQwikTable(options)` causes a vite to crash in dev mode

### ISSUE THREAD

**Issue by thejanasatan** [TanStack/table#5455]
### TanStack Table version

v8.15.3

### Framework/Library version

Qwik 0.0.44

### Describe the bug and the steps to reproduce it

Tried using `@tanstack/qwik-table` in my hobby Qwik/Qwik City project. Vite throws the following error and crashes.

```
QWIK ERROR Code(20) https://github.com/BuilderIO/qwik/blob/main/packages/qwik/src/core/error/error.ts#L28 Error: Code(20) https://github.com/BuilderIO/qwik/blob/main/packages/qwik/src/core/error/error.ts#L28
```

Following the Github link above, it looks related to an incorrect use of `use*` hooks. Linked docs: https://qwik.dev/docs/components/tasks/#use-method-rules

I've attached a Stackblitz environment reproducing the issue.

### Your Minimal, Reproducible Example - (Sandbox Highly Recommended)

https://stackblitz.com/edit/vitejs-vite-r68rwq?file=src%2Froutes%2Findex.tsx

### Do you intend to try to help solve this bug with your own PR?

No, because I do not know how

### Terms & Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I understand that if my bug cannot be reliable reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by thejanasatan** [comment]
One more piece of information - I tried copy-pasting the hook from Github, and it's now working from my copy of the code. Perhaps the issue is with the build? Like something with the Rollup config?

**Comment by KevinVandy** [comment]
Does this only happen in Qwik City?

**Comment by imMadsen** [comment]
No, I tested it using the component library which does not use Qwik City and got the same issue :(

**Comment by yigsvnsla** [comment]
> Does this only happen in Qwik City?

I also try to copy and paste the hook inside my project. and it seems to be working stably...

**Comment by KevinVandy** [comment]
So if I understand correctly, we're only able to currently support a TanStack Qwik Table adapter with certain vite settings.
